WordPress/src/main/java/org/wordpress/android/ui/mysite/jetpackbadge/JetpackAppInstallReceiver.kt

```diff
@@ -0,0 +1,53 @@
+package org.wordpress.android.ui.mysite.jetpackbadge
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.widget.Toast
+import androidx.preference.PreferenceManager
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.push.GCMMessageHandler
+import org.wordpress.android.util.BuildConfigWrapper
+import javax.inject.Inject
+
+class JetpackAppInstallReceiver : BroadcastReceiver() {
+    @Inject lateinit var mBuildConfigWrapper: BuildConfigWrapper
+    @Inject lateinit var gcmMessageHandler: GCMMessageHandler
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        (context?.applicationContext as WordPress).component().inject(this)
+
+        // TODO: Remove this logging before merge
+        StringBuilder().apply {
+            append("Action: ${intent?.action}\n")
+            append("URI: ${intent?.toUri(Intent.URI_INTENT_SCHEME)}\n")
+            toString().also { log ->
+                Log.d(TAG, log)
+                Toast.makeText(context, log, Toast.LENGTH_LONG).show()
+            }
+        }
+
+        if (intent != null && !mBuildConfigWrapper.isJetpackApp) {
```

Comment:
- Should we also check if WP app is logged in? 
- We can add the data of the logged-in account id of JP to broadcast intent and send it to WP. Then we can check if WP account is the same. If it's not the same, we don't disable notifications. Wdyt?

Reply:
Probably not, given this flow is from WP -> to JP and since origin is already in logged in.

Reply:
Build is failing due to lint as it needs a permission defined like so
    `<permission
        android:name="org.wordpress.android.permission.DISABLE_NOTIFICATIONS"
        android:label="@string/notifications_disabled"
        android:description="@string/notifications_disabled"
        android:protectionLevel="normal" />`

and on the `<receiver `
`android:permission="org.wordpress.android.permission.DISABLE_NOTIFICATIONS"`

When add that, not able to use adb command for testing, so left it for now to finish testing
`adb shell am broadcast -p org.wordpress.android.prealpha -a org.wordpress.android.broadcast.DISABLE_NOTIFICATIONS --receiver-permission org.wordpress.android.permission.DISABLE_NOTIFICATIONS`